Using set(VOLK_STATIC_DEFINES VK_USE_PLATFORM_XLIB_KHR) in a CMake project that uses both volk and Jolt Physics causes errors

In a minimal project using both volk (as a static library) and Jolt Physics https://github.com/jrouwe/JoltPhysics, using `set(VOLK_STATIC_DEFINES VK_USE_PLATFORM_XLIB_KHR)` causes errors. The CMake setup I have is:

```
cmake_minimum_required(VERSION 3.20 FATAL_ERROR)

project(jolt_volk)

add_executable(main main.cpp)

set(PHYSICS_REPO_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/JoltPhysics)
include(${PHYSICS_REPO_ROOT}/Jolt/Jolt.cmake)

if (UNIX)
  set(VOLK_STATIC_DEFINES VK_USE_PLATFORM_XLIB_KHR)
endif()
add_subdirectory(volk)

target_link_libraries(main PRIVATE Jolt volk)
```

And my `main.cpp` is:
```
#include <volk.h>

// Includes from Jolt HelloWorld
// https://github.com/jrouwe/JoltPhysics/tree/master/HelloWorld
#include <Jolt/Jolt.h>
#include <Jolt/RegisterTypes.h>
#include <Jolt/Core/Factory.h>
#include <Jolt/Core/TempAllocator.h>
#include <Jolt/Core/JobSystemThreadPool.h>
#include <Jolt/Physics/PhysicsSettings.h>
#include <Jolt/Physics/PhysicsSystem.h>
#include <Jolt/Physics/Collision/Shape/BoxShape.h>
#include <Jolt/Physics/Collision/Shape/SphereShape.h>
#include <Jolt/Physics/Body/BodyCreationSettings.h>
#include <Jolt/Physics/Body/BodyActivationListener.h>

int main() {
    return 0;
}
```

I'm not quite sure what causes this interaction between volk and Jolt and if this is really a problem with volk. But I only get this error when combining the two. Perhaps it has something to do with:
```
  if(VOLK_STATIC_DEFINES)
    target_compile_definitions(volk PUBLIC ${VOLK_STATIC_DEFINES})
```

Reproducer: https://github.com/dry-cat/jolt_volk_reproducer

`make` output: [error.txt](https://github.com/user-attachments/files/18807298/error.txt)

First error from `make` output:
```
In file included from /usr/include/X11/Xlib.h:44,
                 from /usr/include/vulkan/vulkan.h:58,
                 from /home/dry-cat/repos/jolt_volk_reproducer/volk/volk.h:50,
                 from /home/dry-cat/repos/jolt_volk_reproducer/main.cpp:1:
/home/dry-cat/repos/jolt_volk_reproducer/JoltPhysics/Jolt/Physics/Collision/Shape/Shape.h:58:9: error: expected identifier before numeric constant
   58 |         Convex,                                                 ///< Used by ConvexShape, all shapes that use the generic convex vs convex collision detection system (box, sphere, capsule, tapered capsule, cylinder, triangle)
      |         ^~~~~~
/home/dry-cat/repos/jolt_volk_reproducer/JoltPhysics/Jolt/Physics/Collision/Shape/Shape.h:58:9: error: expected ‘}’ before numeric constant
```

OS Name: Ubuntu 22.04.5 LTS, Windowing System: X11
```
-- The C compiler identification is GNU 11.4.0
-- The CXX compiler identification is GNU 11.4.0
```

If this is an issue better suited for the https://github.com/jrouwe/JoltPhysics repo then let me know.
Also, a workaround is using volk as a header-only library.

Anymore info to add @jrouwe?

--- a/volk.h
+++ b/volk.h
@@ -25,31 +25,100 @@
 #ifndef VULKAN_H_
 #	ifdef VOLK_VULKAN_H_PATH
 #		include VOLK_VULKAN_H_PATH
-#	elif defined(VK_USE_PLATFORM_WIN32_KHR)
-#		include <vulkan/vk_platform.h>
-#		include <vulkan/vulkan_core.h>
-
-		/* When VK_USE_PLATFORM_WIN32_KHR is defined, instead of including vulkan.h directly, we include individual parts of the SDK
-		 * This is necessary to avoid including <windows.h> which is very heavy - it takes 200ms to parse without WIN32_LEAN_AND_MEAN
-		 * and 100ms to parse with it. vulkan_win32.h only needs a few symbols that are easy to redefine ourselves.
-		 */
-		typedef unsigned long DWORD;
-		typedef const wchar_t* LPCWSTR;
-		typedef void* HANDLE;
-		typedef struct HINSTANCE__* HINSTANCE;
-		typedef struct HWND__* HWND;
-		typedef struct HMONITOR__* HMONITOR;
-		typedef struct _SECURITY_ATTRIBUTES SECURITY_ATTRIBUTES;
-
-#		include <vulkan/vulkan_win32.h>
-
-#		ifdef VK_ENABLE_BETA_EXTENSIONS
-#			include <vulkan/vulkan_beta.h>
-#		endif
 #	else
-#		include <vulkan/vulkan.h>
-#	endif
+
+/* Instead of directly including vulkan.h, we include individual parts of the SDK
+ * This is necessary to avoid including platform headers (which vulkan.h does unconditionally)
+ * and replace them with forward declarations, which makes build times faster and avoids macro conflicts.
+ */
+#include <vulkan/vk_platform.h>
+#include <vulkan/vulkan_core.h>
+
+#ifdef VK_USE_PLATFORM_ANDROID_KHR
+#include <vulkan/vulkan_android.h>
 #endif
+
+#ifdef VK_USE_PLATFORM_FUCHSIA
+#include <zircon/types.h>
+#include <vulkan/vulkan_fuchsia.h>
+#endif
+
+#ifdef VK_USE_PLATFORM_IOS_MVK
+#include <vulkan/vulkan_ios.h>
+#endif
+
+#ifdef VK_USE_PLATFORM_MACOS_MVK
+#include <vulkan/vulkan_macos.h>
+#endif
+
+#ifdef VK_USE_PLATFORM_METAL_EXT
+#include <vulkan/vulkan_metal.h>
+#endif
+
+#ifdef VK_USE_PLATFORM_VI_NN
+#include <vulkan/vulkan_vi.h>
+#endif
+
+#ifdef VK_USE_PLATFORM_WAYLAND_KHR
+#include <vulkan/vulkan_wayland.h>
+#endif
+
+#ifdef VK_USE_PLATFORM_WIN32_KHR
+typedef unsigned long DWORD;
+typedef const wchar_t* LPCWSTR;
+typedef void* HANDLE;
+typedef struct HINSTANCE__* HINSTANCE;
+typedef struct HWND__* HWND;
+typedef struct HMONITOR__* HMONITOR;
+typedef struct _SECURITY_ATTRIBUTES SECURITY_ATTRIBUTES;
+#include <vulkan/vulkan_win32.h>
+#endif
+
+#ifdef VK_USE_PLATFORM_XCB_KHR
+#include <xcb/xcb.h>
+#include <vulkan/vulkan_xcb.h>
+#endif
+
+#ifdef VK_USE_PLATFORM_XLIB_KHR
+typedef struct _XDisplay Display;
+typedef unsigned long Window;
+typedef unsigned long VisualID;
+#include <vulkan/vulkan_xlib.h>
+#endif
+
+#ifdef VK_USE_PLATFORM_DIRECTFB_EXT
+#include <directfb.h>
+#include <vulkan/vulkan_directfb.h>
+#endif
+
+#ifdef VK_USE_PLATFORM_XLIB_XRANDR_EXT
+typedef struct _XDisplay Display;
+typedef unsigned long RROutput;
+#include <vulkan/vulkan_xlib_xrandr.h>
+#endif
+
+#ifdef VK_USE_PLATFORM_GGP
+#include <ggp_c/vulkan_types.h>
+#include <vulkan/vulkan_ggp.h>
+#endif
+
+#ifdef VK_USE_PLATFORM_SCREEN_QNX
+#include <screen/screen.h>
+#include <vulkan/vulkan_screen.h>
+#endif
+
+#ifdef VK_USE_PLATFORM_SCI
+#include <nvscisync.h>
+#include <nvscibuf.h>
+#include <vulkan/vulkan_sci.h>
+#endif
+
+#ifdef VK_ENABLE_BETA_EXTENSIONS
+#include <vulkan/vulkan_beta.h>
+#endif
+
+#	endif
+#endif /* VULKAN_H_ */
 
 #ifdef __cplusplus
 extern "C" {

--- a/volk.h
+++ b/volk.h
@@ -25,14 +25,97 @@
 #ifndef VULKAN_H_
 #	ifdef VOLK_VULKAN_H_PATH
 #		include VOLK_VULKAN_H_PATH
-#	else
+#	else /* Platform headers included below */
+#		include <vulkan/vk_platform.h>
+#		include <vulkan/vulkan_core.h>
+#	endif
+#endif
 
-/* Instead of directly including vulkan.h, we include individual parts of the SDK
- * This is necessary to avoid including platform headers (which vulkan.h does unconditionally)
- * and replace them with forward declarations, which makes build times faster and avoids macro conflicts.
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct VolkDeviceTable;
+
+/**
+ * Initialize library by loading Vulkan loader; call this function before creating the Vulkan instance.
+ *
+ * Returns VK_SUCCESS on success and VK_ERROR_INITIALIZATION_FAILED otherwise.
  */
-#include <vulkan/vk_platform.h>
-#include <vulkan/vulkan_core.h>
+VkResult volkInitialize(void);
+
+/**
+ * Initialize library by providing a custom handler to load global symbols.
+ *
+ * This function can be used instead of volkInitialize.
+ * The handler function pointer will be asked to load global Vulkan symbols which require no instance
+ * (such as vkCreateInstance, vkEnumerateInstance* and vkEnumerateInstanceVersion if available).
+ */
+void volkInitializeCustom(PFN_vkGetInstanceProcAddr handler);
+
+/**
+ * Finalize library by unloading Vulkan loader and resetting global symbols to NULL.
+ *
+ * This function does not need to be called on process exit (as loader will be unloaded automatically) or if volkInitialize failed.
+ * In general this function is optional to call but may be useful in rare cases eg if volk needs to be reinitialized multiple times.
+ */
+void volkFinalize(void);
+
+/**
+ * Get Vulkan instance version supported by the Vulkan loader, or 0 if Vulkan isn't supported
+ *
+ * Returns 0 if volkInitialize wasn't called or failed.
+ */
+uint32_t volkGetInstanceVersion(void);
+
+/**
+ * Load global function pointers using application-created VkInstance; call this function after creating the Vulkan instance.
+ */
+void volkLoadInstance(VkInstance instance);
+
+/**
+ * Load global function pointers using application-created VkInstance; call this function after creating the Vulkan instance.
+ * Skips loading device-based function pointers, requires usage of volkLoadDevice afterwards.
+ */
+void volkLoadInstanceOnly(VkInstance instance);
+
+/**
+ * Load global function pointers using application-created VkDevice; call this function after creating the Vulkan device.
+ *
+ * Note: this is not suitable for applications that want to use multiple VkDevice objects concurrently.
+ */
+void volkLoadDevice(VkDevice device);
+
+/**
+ * Return last VkInstance for which global function pointers have been loaded via volkLoadInstance(),
+ * or VK_NULL_HANDLE if volkLoadInstance() has not been called.
+ */
+VkInstance volkGetLoadedInstance(void);
+
+/**
+ * Return last VkDevice for which global function pointers have been loaded via volkLoadDevice(),
+ * or VK_NULL_HANDLE if volkLoadDevice() has not been called.
+ */
+VkDevice volkGetLoadedDevice(void);
+
+/**
+ * Load function pointers using application-created VkDevice into a table.
+ * Application should use function pointers from that table instead of using global function pointers.
+ */
+void volkLoadDeviceTable(struct VolkDeviceTable* table, VkDevice device);
+
+#ifdef __cplusplus
+}
+#endif
+
+/* Instead of directly including vulkan.h, we include platform-specific parts of the SDK manually
+ * This is necessary to avoid including platform headers in some cases (which vulkan.h does unconditionally)
+ * and replace them with forward declarations, which makes build times faster and avoids macro conflicts.
+ *
+ * Note that we only replace platform-specific headers when the headers are known to be problematic: very large
+ * or slow to compile (Windows), or introducing unprefixed macros which can cause conflicts (Windows, Xlib).
+ */
+#if !defined(VULKAN_H_) && !defined(VOLK_VULKAN_H_PATH)
 
 #ifdef VK_USE_PLATFORM_ANDROID_KHR
 #include <vulkan/vulkan_android.h>
@@ -117,81 +200,7 @@ typedef unsigned long RROutput;
 #include <vulkan/vulkan_beta.h>
 #endif
 
-#	endif
-#endif /* VULKAN_H_ */
-
-#ifdef __cplusplus
-extern "C" {
 #endif
-
-struct VolkDeviceTable;
-
-/**
- * Initialize library by loading Vulkan loader; call this function before creating the Vulkan instance.
- *
- * Returns VK_SUCCESS on success and VK_ERROR_INITIALIZATION_FAILED otherwise.
- */
-VkResult volkInitialize(void);
-
-/**
- * Initialize library by providing a custom handler to load global symbols.
- *
- * This function can be used instead of volkInitialize.
- * The handler function pointer will be asked to load global Vulkan symbols which require no instance
- * (such as vkCreateInstance, vkEnumerateInstance* and vkEnumerateInstanceVersion if available).
- */
-void volkInitializeCustom(PFN_vkGetInstanceProcAddr handler);
-
-/**
- * Finalize library by unloading Vulkan loader and resetting global symbols to NULL.
- *
- * This function does not need to be called on process exit (as loader will be unloaded automatically) or if volkInitialize failed.
- * In general this function is optional to call but may be useful in rare cases eg if volk needs to be reinitialized multiple times.
- */
-void volkFinalize(void);
-
-/**
- * Get Vulkan instance version supported by the Vulkan loader, or 0 if Vulkan isn't supported
- *
- * Returns 0 if volkInitialize wasn't called or failed.
- */
-uint32_t volkGetInstanceVersion(void);
-
-/**
- * Load global function pointers using application-created VkInstance; call this function after creating the Vulkan instance.
- */
-void volkLoadInstance(VkInstance instance);
-
-/**
- * Load global function pointers using application-created VkInstance; call this function after creating the Vulkan instance.
- * Skips loading device-based function pointers, requires usage of volkLoadDevice afterwards.
- */
-void volkLoadInstanceOnly(VkInstance instance);
-
-/**
- * Load global function pointers using application-created VkDevice; call this function after creating the Vulkan device.
- *
- * Note: this is not suitable for applications that want to use multiple VkDevice objects concurrently.
- */
-void volkLoadDevice(VkDevice device);
-
-/**
- * Return last VkInstance for which global function pointers have been loaded via volkLoadInstance(),
- * or VK_NULL_HANDLE if volkLoadInstance() has not been called.
- */
-VkInstance volkGetLoadedInstance(void);
-
-/**
- * Return last VkDevice for which global function pointers have been loaded via volkLoadDevice(),
- * or VK_NULL_HANDLE if volkLoadDevice() has not been called.
- */
-VkDevice volkGetLoadedDevice(void);
-
-/**
- * Load function pointers using application-created VkDevice into a table.
- * Application should use function pointers from that table instead of using global function pointers.
- */
-void volkLoadDeviceTable(struct VolkDeviceTable* table, VkDevice device);
 
 /**
  * Device-specific function pointer table
@@ -1123,6 +1132,10 @@ struct VolkDeviceTable
 #endif /* (defined(VK_KHR_device_group) && defined(VK_KHR_swapchain)) || (defined(VK_KHR_swapchain) && defined(VK_VERSION_1_1)) */
 	/* VOLK_GENERATE_DEVICE_TABLE */
 };
+
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 /* VOLK_GENERATE_PROTOTYPES_H */
 #if defined(VK_VERSION_1_0)

